src/Halcyon/HAL/HALResponse.cs

```diff
@@ -26,9 +25,9 @@ public class HALResponse {
         public HALResponse(object model, IHALModelConfig config = null)
             : this(config) {
             if (!(model is JObject) && (model is IEnumerable)) {
-                throw new ArgumentException("The HAL model should be Enumerable. You should use an embedded collection instead", "model");
+                throw new ArgumentException("The HAL model should be Enumerable. You should use an embedded collection instead", nameof(model));
             }
-
+            HALAttributeResolver.ResolveAttributes(this, model);
```

Comment:
I don't like the circular dependency this creates between the `HALResponse` and `HALAttributeResolver`. The resolver code is impossible to isolate and in the `HALResponse`.

I don't think resolving the attributes should be the responsibility of the `HALResponse`. 

I think there should be an `IHALConverter` interface, (open to a better name), that is optionally used by the Formatters to convert a plain model to a `HALResponse`.

````c#
public interface IHALConverter {
  bool CanConvert(object model);
  HALResponse GetHALResponse(object model);
}

public JsonHALMediaTypeFormatter(string[] halJsonMediaTypes = null, string[] jsonMediaTypes = null, IHALConverter converter = null) {
    ...
    this.converter = converter ?? new HALAttributeConverter();
    ...
}
````

Does this make sense to you?